google-cloud-logging/lib/google/cloud/logging/project.rb

```diff
@@ -332,11 +332,12 @@ def async_writer max_queue_size: AsyncWriter::DEFAULT_MAX_QUEUE_SIZE
         #   resource to be associated with written log entries.
         # @param [Hash] labels A set of user-defined data to be associated with
         #   written log entries.
-        # @param [Boolean|AsyncWriter] async_writer An AsyncWriter for the
-        #   logger to transmit log entries. You may also pass true to request
-        #   a new AsyncWriter for this logger, or false to request no
-        #   AsyncWriter (which will cause the logger to make blocking calls).
-        #   Default is true.
+        # @param [Boolean] async Whether entries are written asynchronously.
+        #   When `true` the logger will use use a new AsyncWriter instance. When
+        #   `false` it will write entries synchronously using
+        #   Project#write_entries. Default is true.
+        # @param [AsyncWriter|#write_entries] writer The object to write entries
```

Comment:
1. Every AsyncWriter manages exactly 1 background thread. So there's a common use case for providing the AsyncWriter for a Logger: if you create N Loggers and you want them all to share the same writer. (This is actually what we expect to do for framework integration: we're creating a separate Logger for each request, since each will need to be configured with that request's trace ID, but we want them all to share the same AsyncWriter.)
2. That's a good question. I think you could argue that it should always be async, since users of Logger would probably never expect calls to block or return errors. I guess the main reason I wanted the option is it just makes me uneasy to force a Ruby application to use threads. But there's ActiveRecord and its thread pool, so maybe I'm just barking at nothing.